cpp_linter/rest_api/github_api.py

```diff
@@ -522,7 +650,125 @@ def _dismiss_stale_reviews(self, url: str):
                         url=f"{url}/{review['id']}/dismissals",
                         method="PUT",
                         data=json.dumps(
-                            {"message": "outdated suggestion", "event": "DISMISS"}
+                            {"message": "Outdated review", "event": "DISMISS"}
                         ),
                         strict=False,
                     )
+
+    def _get_existing_review_comments(self, no_dismissed: bool = True):
+        """Creates the list existing review thread comments to close.
+
+        :param no_dismissed: `True` to ignore any already dismissed comments.
+        """
+        repo_owner, repo_name = self.repo.split("/")
+        query = QUERY_REVIEW_COMMENTS % (
+            repo_owner,
+            repo_name,
+            self.pull_request,
+        )
+        response = self.api_request(
+            url=f"{self.api_url}/graphql",
+            method="POST",
+            data=json.dumps({"query": query}),
+            strict=False,
+        )
+        if response.status_code != 200:
+            logger.error(
+                "Could not get existing review thread comments: %d", response.status_code
+            )
+            return
+        data = response.json()
+        found_threads = []
+        nodes = data["data"]["repository"]["pullRequest"]["reviewThreads"]["nodes"]
+        for thread in nodes:
+            for comment in thread["comments"]["nodes"]:
+                if (
+                    comment["id"]
+                    and (
+                        not no_dismissed
+                        or (
+                            thread["isResolved"] is False
+                            and thread["isCollapsed"] is False
+                        )
+                    )
+                    and comment["author"]["login"] == "github-actions"
+                    and (
+                        comment["body"].strip().startswith("### clang-format")
+                        or comment["body"].strip().startswith("### clang-tidy")
+                    )
```

Comment:
> ```py
> comment["author"]["login"] == "github-actions"
> ```

This won't always be true. Users could pass a PAT (Personal Access Token) instead of the `secrets.github-token`, effectively uses cpp-linter as part of a bot account...

Instead, we usually tag the start of our generated comments with an html comment: https://github.com/maoliver-amd/cpp-linter/blob/0125f5935857d8b131db1f092ec9dd3b2a9cbf7c/cpp_linter/rest_api/__init__.py#L22

We should also start prefixing our review comments' suggestions with this marker as well. This way, we can replace
```py
> and comment["author"]["login"] == "github-actions" 
> and (
>    comment["body"].strip().startswith("### clang-format")
>    or comment["body"].strip().startswith("### clang-tidy")
> )
```
with
```py
and comment["body"].startswith(COMMENT_MARKER)
```

Reply:
submitted #132

Reply:
#132 was merged. You can rebase, then just
```suggestion
                    and comment["body"].startswith(COMMENT_MARKER)
```